### PR TITLE
Implement WriteIndex for cluster summaries

### PR DESCRIPTION
**Pull request by charlieegan3**
part of #54

### CODE REVIEW

File: cmd/check.go [hunk 1]
```diff
@@ -338,6 +341,13 @@ func check() {
 			log.Fatalf("Cannot marshal intermediate result: %v", err)
 		}
 
+		// build a report to build the updated context for the report index
+		report, err := reports.NewReport(manifest, rc)
```
Comment (j-fuentes) [inline cmd/check.go]: Possible optimization: We don't really need the whole report here. We could create here the `ReportSummary`.

But that's is not a reason to block this PR.

File: pkg/output/azblob/azblob.go [hunk 2]
```diff
@@ -78,3 +78,26 @@ func (o *Output) Write(ctx context.Context, policyManifest *packaging.PolicyMani
...
+	blobURL := containerURL.NewBlockBlobURL(fmt.Sprintf("index/%s%s", cluster, o.exporter.FileExtension()))
+
+	_, err = azblob.UploadStreamToBlockBlob(ctx, buffer, blobURL, azblob.UploadStreamToBlockBlobOptions{
+		// values chosen arbitrarily
+		BufferSize: 2 * 1024 * 1024,
+		MaxBuffers: 3,
+	})
+	if err != nil {
+		return err
+	}
```
Comment (j-fuentes) [inline pkg/output/azblob/azblob.go]: This is very similar to what is done in `Write`, I'd create a common function.

File: cmd/check.go (hunk 1 above)
Comment (charlieegan3) [reply cmd/check.go]: Yeah... I did wonder about this... it's not ideal.

File: pkg/output/azblob/azblob.go (hunk 2 above)
Comment (charlieegan3) [reply pkg/output/azblob/azblob.go]: Yeah, that fair. I've 'DRYed' this up in 688e94ca4f091b2520d7eb074de875e78681e693

**Comment by charlieegan3** [comment]
I had to rebase and fix a failed test...

**Pull request merged**